neuralprophet/configure.py

```diff
@@ -161,19 +171,41 @@ def set_auto_batch_epoch(
         # also set lambda_delay:
         self.lambda_delay = int(self.reg_delay_pct * self.epochs)
 
-    def get_optimizer(self, model_parameters):
-        return utils_torch.create_optimizer_from_config(self.optimizer, model_parameters, self.learning_rate)
-
-    def get_scheduler(self, optimizer, steps_per_epoch):
-        return torch.optim.lr_scheduler.OneCycleLR(
-            optimizer,
-            max_lr=self.learning_rate,
-            epochs=self.epochs,
-            steps_per_epoch=steps_per_epoch,
-            pct_start=0.3,
-            anneal_strategy="cos",
-            div_factor=100.0,
-            final_div_factor=5000.0,
+    def set_optimizer(self):
+        """
+        Set the optimizer and optimizer args. If optimizer is a string, then it will be converted to the corresponding torch optimizer.
+        The optimizer is not initialized yet as this is done in configure_optimizers in TimeNet.
+        """
+        self.optimizer, self.optimizer_args = utils_torch.create_optimizer_from_config(
+            self.optimizer, self.optimizer_args
+        )
+
+    def set_scheduler(self):
+        """
+        Set the scheduler and scheduler args.
+        The scheduler is not initialized yet as this is done in configure_optimizers in TimeNet.
+        """
+        self.scheduler = torch.optim.lr_scheduler.OneCycleLR
+        self.scheduler_args.update(
+            {
+                "pct_start": 0.3,
+                "anneal_strategy": "cos",
+                "div_factor": 100.0,
+                "final_div_factor": 5000.0,
+            }
+        )
+
+    def set_lr_finder_args(self, dataset_size):
+        """
+        Set the lr_finder_args.
+        This is the range of learning rates to test.
+        """
+        self.lr_finder_args.update(
+            {
+                "min_lr": 1e-7,
+                "max_lr": 100,
+                "num_training": 50 + int(np.log10(100 + dataset_size) * 25),
+            }
```

Comment:
@ourownstory Currently this does not allow the user to overwrite the default `min_lr` etc., should we allow the user to change it or are we fine with keeping it static for now?